src/Microsoft.SqlTools.ServiceLayer/Metadata/MetadataService.cs

```diff
@@ -153,27 +153,47 @@ internal static void GenerateServerContextualization(GenerateServerContextualiza
                 {
                     // If scripts have been generated within the last 30 days then there isn't a need to go through the process
                     // of generating scripts again.
-                    if (!MetadataScriptTempFileStream.IsScriptTempFileUpdateNeeded(connectionInfo.ConnectionDetails.ServerName))
-                    {
-                        return;
-                    }
-
-                    var scripts = SmoScripterHelpers.GenerateAllServerTableScripts(sqlConn);
-                    if (scripts != null)
+                    if (MetadataScriptTempFileStream.IsScriptTempFileUpdateNeeded(connectionInfo.ConnectionDetails.ServerName))
                     {
-                        try
+                        var scripts = SmoScripterHelpers.GenerateAllServerTableScripts(sqlConn);
+                        if (scripts != null)
                         {
-                            MetadataScriptTempFileStream.Write(connectionInfo.ConnectionDetails.ServerName, scripts);
+                            try
+                            {
+                                MetadataScriptTempFileStream.Write(connectionInfo.ConnectionDetails.ServerName, scripts);
+                            }
+                            catch (Exception ex)
+                            {
+                                Logger.Error($"An error was encountered while writing to the cache. Error: {ex.Message}");
+
+                                var contextFailedToCompleteParams = new GenerateServerContextualizationCompleteParams()
+                                {
+                                    OwnerUri = contextualizationParams.OwnerUri,
+                                    ErrorMessage = ex.Message
```

Comment:
is `CompletedGeneratingContext` expected to be false here? It would be clearer when reading the code to explicitly set it